tests/LSP17Extensions/Extension4337/4337.test.ts

```diff
@@ -0,0 +1,224 @@
+
+    keyManager = await new LSP6KeyManager__factory(deployer).deploy(universalProfile.address);
+
+    // transfer ownership to keyManager
+    await universalProfile.transferOwnership(keyManager.address);
+
+    const dataKey =
+      ERC725YDataKeys.LSP6['AddressPermissions:Permissions'] + deployerAddress.slice(2);
+
+    await universalProfile.setData(dataKey, ALL_PERMISSIONS);
+
+    const acceptOwnershipBytes = universalProfile.interface.encodeFunctionData('acceptOwnership');
+    await keyManager.execute(acceptOwnershipBytes);
+    expect(await universalProfile.owner()).to.eq(keyManager.address);
+
+    // deploy entrypoint
+    entryPoint = await deployEntryPoint();
+    expect(await isDeployed(entryPoint.address)).to.eq(true);
+
+    // give all permissions to entrypoint
+    const dataKeyEntryPointPermissions =
+      ERC725YDataKeys.LSP6['AddressPermissions:Permissions'] + entryPoint.address.slice(2);
+    await universalProfile.setData(dataKeyEntryPointPermissions, ALL_PERMISSIONS);
+
+    // deploy extension and attach it to universalProfile
+    const extension4337 = await new Extension4337__factory(deployer).deploy(entryPoint.address);
+    const validateUserOpSigHash = extension4337.interface.getSighash('validateUserOp');
+
+    const extensionDataKey =
+      ERC725YDataKeys.LSP17.LSP17ExtensionPrefix +
+      validateUserOpSigHash.slice(2) +
+      '00000000000000000000000000000000';
+
+    await universalProfile.setData(extensionDataKey, extension4337.address);
+
+    // give permissions to controllers
+    const dataKeyWithPermission4337 =
+      ERC725YDataKeys.LSP6['AddressPermissions:Permissions'] +
+      controllerWith4337Permission.address.slice(2);
+    await universalProfile.setData(
+      dataKeyWithPermission4337,
+      combinePermissions(ALL_PERMISSIONS, Permission4337),
+    );
+
+    const dataKeyWithoutPermission4337 =
+      ERC725YDataKeys.LSP6['AddressPermissions:Permissions'] +
+      controllerWithout4337Permission.address.slice(2);
+    await universalProfile.setData(dataKeyWithoutPermission4337, ALL_PERMISSIONS);
+
+    const dataKeyWithOnlyPermission4337 =
+      ERC725YDataKeys.LSP6['AddressPermissions:Permissions'] +
+      controllerWithOnly4337Permission.address.slice(2);
+
+    await universalProfile.setData(dataKeyWithOnlyPermission4337, Permission4337);
+
+    // execute call data
+    transferCallData = universalProfile.interface.encodeFunctionData('execute', [
+      0,
+      ethers.constants.AddressZero,
+      amountToTransfer,
+      '0x1234',
+    ]);
+
+    // send 1 ethers to universalProfile
+    await deployer.sendTransaction({
+      to: universalProfile.address,
+      value: parseEther('1'),
+    });
+
+    // stake on entrypoint
+    const stakeAmount = parseEther('1');
+    await entryPoint.depositTo(universalProfileAddress, { value: stakeAmount });
+  });
+
+  it('should pass', async function () {
+    const address0BalanceBefore = await getBalance(ethers.constants.AddressZero);
```

Comment:
Suggested as easier to read (hard to see the `0` in the constant name).

```suggestion
    const addressZeroBalanceBefore = await getBalance(ethers.constants.AddressZero);
```